### ISSUE TITLE
SapientGuardian + ASP Net Core 2.0 preview

### ISSUE THREAD

**Issue by mkilp** [SapientGuardian/SapientGuardian.EntityFrameworkCore.MySql#56]
Heya!

I'm planning on using your Framework in a ASP Net Core 2.0 preview project.

Sadly on running the project i'm encountering this error:

> System.TypeLoadException: "Method 'Clone' in type 'MySQL.Data.Entity.MySQLOptionsExtension' from assembly 'SapientGuardian.EntityFrameworkCore.MySql, Version=7.1.21.0, Culture=neutral, PublicKeyToken=null' does not have an implementation."

I could implement the function myself, but sadly I don't have a clue where.

Any Ideas?

Thanks,
Marvin

**Comment by SapientGuardian** [comment]
Did you jump Microsoft.EntityFrameworkCore to a newer version? It looks like that Clone method was added, abstract, in https://github.com/aspnet/EntityFramework/commit/09b1503ef94287784d1f8fb29da90cba2a891f5e#diff-87e1640e3a0196093b5f92e903872b28R31.

**Comment by mkilp** [comment]
Aww, looks like it. 
Tried going for the signalR preview, but was not able to get it running sadly. And yeah I seemed to have needed to jump up the EntityFrameworkCore Version, because of dependencies.

**Comment by labiraus** [comment]
Has anyone got a work around for this issue?

**Comment by SapientGuardian** [comment]
Now that .net core 2 has been release in preview, I took a look to see what it would take to upgrade this library to link against EF Core 2, and it turns out to be a _lot_ of work.  Far more than just this particular missing method, there are entire classes that have been removed in favor of providers implementing their own, on top of some other minor changes in method signatures that would need to be addressed. And even if all of that work was done, it wouldn't fix any of the outstanding bugs we currently have logged for this project.

In short, the only way this library is going to support EF Core 2 is if someone submits a PR to do it. While it doesn't look like the Pomelo folks are supporting 2.0 yet either, they are much more active in maintaining their library, so I'd keep an eye on them and consider switching over.
